src/app/modules/shared-components/components/input/input.component.ts

```diff
@@ -24,21 +23,28 @@ export class InputComponent implements OnInit, OnChanges {
   @Input() inputType = 'text';
   @Input() inputIcon = 'font'; // a font-awesome icon identifier
   @Input() buttonIcon?: string; // a font-awesome icon identifier for the input button
+  @Input() hasClearButton = false;
 
   @Output() click = new EventEmitter();
 
-  control: AbstractControl | null = null;
+  subscription?: Subscription;
+  value = '';
```

Comment:
what do you use this value for?

Reply:
Use to verify that the length of the input and show the 'x' if there is something. But I've removed it to use a template variable